automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala

```diff
@@ -0,0 +1,194 @@
+
+trait TestMethods {
+  /* METHODS */
+
+  val SubWorkflowChildMethod = Method(
+    methodName = "child_method_for_subworkflow_test",
+    methodNamespace = "subworkflow_methods",
+    snapshotId = 1,
+    rootEntityType = "participant",
+    synopsis = "testtestsynopsis",
+    documentation = "",
+    payload =
+      """
+        |workflow turboHC {
+        |
+        |	File refFasta
+        |	File refIndex
+        |	File refDict
+        |
+        |	File inputBam
+        |	File inputBamIndex
+        |	File intervalsList
+        |
+        |	String gatk_docker
+        |
+        |	String output_name = basename(inputBam, ".bam") + ".g.vcf"
+        |
+        |	Array[String] callingIntervals = read_lines(intervalsList)
+        |
+        |	scatter(interval in callingIntervals) {
+        |		call HaplotypeCaller_GVCF {
+        |			input:
+        |				refFasta = refFasta,
+        |				refIndex = refIndex,
+        |				refDict = refDict,
+        |
+        |				inputBam = inputBam,
+        |				inputBamIndex = inputBamIndex,
+        |				intervals = interval,
+        |				gvcf_name = output_name,
+        |				docker_image = gatk_docker
+        |		}
+        |	}
+        |	call MergeGVCFs {
+        |		input:
+        |			gvcfs = HaplotypeCaller_GVCF.output_gvcf,
+        |			gvcf_name = output_name,
+        |			docker_image = gatk_docker
+        |	}
+        |
+        |	output {
+        |		File outputGVCF = MergeGVCFs.mergedGVCF
+        |	}
+        |}
+        |
+        |task HaplotypeCaller_GVCF {
+        |
+        |	File refFasta
+        |	File refIndex
+        |	File refDict
+        |	File inputBam
+        |	File inputBamIndex
+        |
+        |	String gvcf_name
+        |	String intervals
+        |
+        |	String docker_image
+        |
+        |	command {
+        |		gatk HaplotypeCaller \
+        |			-R ${refFasta} \
+        |			-I ${inputBam} \
+        |			-O ${gvcf_name} \
+        |			-ERC GVCF \
+        |			-L ${intervals} \
+        |			-ip 100
+        |	}
+        |
+        |	output {
+        |		File output_gvcf = "${gvcf_name}"
+        |	}
+        |
+        |	runtime {
+        |		docker: docker_image
+        |	}
+        |}
+        |
+        |task MergeGVCFs {
+        |
+        |	Array[File] gvcfs
+        |	String gvcf_name
+        |
+        |	String docker_image
+        |
+        |  command {
+        |    gatk MergeVcfs \
+        |	    -I ${sep=' -I' gvcfs} \
+        |	    -O ${gvcf_name}
+        |  }
+        |
+        |  output {
+        |    File mergedGVCF = "${gvcf_name}"
+        |  }
+        |
+        |  runtime {
+        |	docker: docker_image
+        |  }
+        |
+        |}
+      """.stripMargin)
+
+  def subWorkflowParentMethod(child: Method): Method = {
+    val childUrl = s"http://orch-app:8080/ga4gh/v1/tools/${child.methodNamespace}:${child.methodName}/versions/1/plain-WDL/descriptor"
```

Comment:
woah. cool that this works

Reply:
this took many iterations.

...but now it occurs to me that this is not likely to work on Alpha.  Hrm.

Reply:
Use the mock server? That’s what I did for Dockstore (search around and you’ll find it)

Reply:
I'd rather not use mocks in an Integration Test.

Reply:
Trying a new version that uses this for FIABs but the proper Orch API URL for real envs

Reply:
Success!  Ran on Alpha and failed in the correct way (import succeeded, and there were no subWorkflows in metadata.  Need this to merge for that to pass.)